src/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -341,6 +350,50 @@ public static byte[] ReadAllBytes(string path)
             }
         }
 
+        private static byte[] ReadAllBytesUnknownLength(FileStream fs)
+        {
+            byte[] rentedArray = null;
+            Span<byte> buffer = stackalloc byte[512];
+            try
+            {
+                int bytesRead = 0;
+                while (true)
+                {
+                    if (bytesRead == buffer.Length)
+                    {
+                        uint newLength = (uint)buffer.Length * 2;
+                        if (newLength > MaxByteArrayLength)
+                        {
+                            newLength = (uint)Math.Max(MaxByteArrayLength, buffer.Length + 1);
```

Comment:
Can't there be signed integer overflow in `buffer.Length + 1` at some point? We don't seem to have any hard limit for this whole loop. Maybe you actually meant `Math.Min` here?

Reply:
For `buffer.Length + 1 > MaxByteArrayLength` case, Rent will end up throwing (I think).

Reply:
Do I understand it correctly - when buffer.Length becomes MaxByteArrayLength on all further iterations it grows by one byte?

Reply:
@tmds That's the thing - DefaultArrayPool.Rent implementation doesn't seem have any limits (or maybe I missed one)

Reply:
From what I see this is how it behaves:
On each iteration size doubles;
Then it tries MaxByteArrayLength;
And on the next iteration (buffer.Length + 1), Rent throws.

Reply:
So I'm looking into DefaultArrayPool.Rent implementation and in case requested length is greater than supported by pool it just creates an array of requested length:
https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs#L108

Reply:
So newermind. I didn't realize that MaxArrayLength is the actual maximum supported by .NET